core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -111,6 +112,14 @@
         Setting.byteSizeSetting("index.translog.flush_threshold_size", new ByteSizeValue(512, ByteSizeUnit.MB), Property.Dynamic,
             Property.IndexScope);
 
+    public static final Setting<ByteSizeValue> INDEX_TRANSLOG_RETENTION_SIZE_SETTING =
```

Comment:
Between size and age you have an inconsistent ordering, sometimes size then age, sometimes age then size. Can you please pick a consistent ordering for the pairing?